docs/external-node.md

```diff
@@ -0,0 +1,532 @@
+machine or a bare-metal server in this document.
+
+Antrea NetworkPolicy is applied to an external Node via the `ExternalEntity`
+resource. `antrea-controller` creates an `ExternalEntity` resource for each
+network interface defined in the `ExternalNode` resource.
+
+`antrea-agent` is running on the external Node, and it controls network
+connectivity and security by attaching the network interface(s) to an OVS bridge.
+A new OpenFlow pipeline has been implemented, dedicated to the ExternalNode
+feature.
+
+You may be interested in using this capability for the below scenarios:
+
+- To apply Antrea NetworkPolicy to an external Node.
+- You want the same security configurations on the external Node for all
+  Operating Systems.
+
+This guide demonstrates how to configure `ExternalNode` to achieve the above
+result.
+
+# Prerequisites
+
+`ExternalNode` is introduced in v1.8 as an alpha feature. A feature gate
+`ExternalNode` must be enabled on the `antrea-controller` and `antrea-agent`in
+the configuration. The configuration for `antrea-controller` is modified in the
+`antrea-config` ConfigMap as follows for the feature to work:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: antrea-config-dcfb6k2hkm
+  namespace: kube-system
+data:
+  antrea-controller.conf: |
+    featureGates:
+      ExternalNode: true
+```
+
+Service `antrea` is created for `antrea-controller` to accept connections from
+`antrea-agent`, and the type is `ClusterIP` by default. Since an external Node
+is running outside the Kubernetes cluster, the `ClusterIP` address is not
+accessible. So a `NodePort` or `LoadBalancer` typed Service for `antrea` is
+required.
+
+Since `antrea-agent` is running on an external Node which is not managed by
+Kubernetes, a single configuration file is required to locate on each machine
+where `antrea-agent` is running. 
+
+For the detail configuration options when running `antrea-agent`on an external
+Node, please refer to the [sample configuration file](../build/yamls/externalnode/conf/antrea-agent.conf).
+
+# The ExternalNode resource
+
+A typical `ExternalNode` resource example:
+
+```yaml
+apiVersion: crd.antrea.io/v1alpha1
+kind: ExternalNode
+metadata:
+  name: vm1
+  namespace: vm-ns
+  labels:
+    role: db
+spec:
+  interfaces:
+    - ips: [ "172.16.100.3" ]
+      name: ""
+```
+
+## Name and Namespace
+
+The `name` field in an `ExternalNode` is unique, which is the key to tell
+`antrea-agent` where it is running on. It is also used as the key in the
+connection from `antrea-agent` to `antrea-controller`. The `ExternalNode` name
+is given to `antrea-agent` via an environment variable `NODE_NAME`, and
+`antrea-agent` will use the hostname to find the `ExternalNode` resource if it
+is not set.
+
+`ExternalNode` resource is working in `namespace` scope. The `namespace` is
+given to `antrea-agent` with option `externalNodeNamespace` in [antrea-agent.conf](../build/yamls/externalnode/conf/antrea-agent.conf).
```

Comment:
The `namespace` name is provided to the `antrea-agent`

BTW usually write "Namespace", not `namespace`, in our documentation

Reply:
Updated in this document.